### ISSUE TITLE
Incomplete item stacking/display in 0.3.1

### ISSUE THREAD

**Issue by graeye** [badasintended/slotlink#10]
**Setup**

- Minecraft 1.16.1
- Fabric API - fabric-api-0.14.0+build.371-1.16.jar
- Fabric Language Kotlin - fabric-language-kotlin-1.3.71+build.1.jar
- slotlink - slotlink-0.3.1+minecraft-1.16.1.jar
- Spinnery - spinnery-3.0.50+fabric-1.16.x.jar
- Pack: All of Fabric 3, modified.

**Problem**

- Large collections of items no longer display as a single entry in storage, and instead display as multiple stacks of 128 items and then some seemingly random partial stacks.

**Reproduction**

- I used an existing storage network from slotlink 0.2.7 and spinnery 3.0.48 with roughly 3,000 cobblestone inside. I can freely move back and forth between version pairs and repeat the results, without changing anything inside the storage system or the system pieces themselves.

**Remarks**

- With slotlink 0.2.7 and spinnery 3.0.48, I see a single Cobblestone item in storage with 3k count.
- With slotlink 0.3.1 and spinnery 3.0.50, I see 22 stacks of 128 Cobblestone. Then a stack of 64, a stack of 63, a stack of 45 and a stack of 30.
- Your release notes mention spinnery 3.0.51, but I do not see that version available on CurseForge.

I also have a crash when attempting to move items from REI, but I'll add that as it's own issue. Thanks for your efforts.

**Comment by deirn** [comment]
What if you disable spinnery 3.0.50 altogether since slotlink already included 3.0.51 in the jar?

**Comment by deirn** [comment]
yeah this seems because of 3.0.50

**Comment by milkev** [comment]
removing the spinnery from my mod folder (since i had downloaded it manually) fixed this.

**Comment by deirn** [comment]
Yeah, looks like 3.0.50 changed how itemstack comparing work and 3.0.51 revert it.
I'll keep this issue open until vini released 3.0.51+ on CF.

**Comment by graeye** [comment]
Confirmed that removing 3.0.50 and not manually having Spinnery in /mods fixed the stack count/display issue.

**Comment by TheVillageid10t** [comment]
So i did diern's fix from issue #11 :

Added "badasintended.slotlink.client.gui.screen.RequestScreen" on .minecraft/config/inventorysorter/blacklist.json5 and its working.
There's not much I can do with this, sorry.

And then tried this removing spinnery to fix the item stacking issue and this force close happens about a second after accessing a network:

https://paste.ee/p/AmtaO

I'll just deal with the items not stacking for now and leave spinnery in there just to have the game playable.

**Comment by deirn** [comment]
hmm, looks like a problem with exotic-matter stuff, like exotic blocks. I'll check it

**Comment by deirn** [comment]
@TheVillageid10t can you tell me what Exotic Blocks items you have in the network?

**Comment by TheVillageid10t** [comment]
I appreciate it. And also tried running it again and idk if it helps at all or not but I can open the gui for the network but as soon as i start scrolling through that's when it decides to exit the game. had the world loaded for about 10 minutes without touching the network (ran around and did other stuff). Go to just open the gui for the network and let it sit to see if it closes, after about another ten minutes I scroll through the network and then it closes the game.

**Comment by deirn** [comment]
yeah it seems like the slot failed to render some xb items

**Comment by TheVillageid10t** [comment]
> @TheVillageid10t can you tell me what Exotic Blocks items you have in the network?

Sure thing give me a sec i'll swap out for 0.2.7 so i can at least see what's in there without it crashing

**Comment by TheVillageid10t** [comment]
So for some reason i couldn't swap back to the old 0.2.7 but i managed to just open up  and look though all the chests with 0.3.1 without any issues. The Exotic blocks in the network are; Sandstone Flat Panel Prismarine Lamp (110 blocks in total) a stack of 64 in one double chest and another stack of 46 in a different double chest)

**Comment by deirn** [comment]
thanks, i'll look to it

**Comment by TheVillageid10t** [comment]
> thanks, i'll look to it

Went though and removed all of the "Exotic Blocks" blocks from the network. Solves the issue. and then re-logged with spinnery 3.0.50 removed and item stack issue gets resolved. Also i think i gained about 20 more fps, before all this I was getting around 65-75 fps with the fixes i'm stabilizing around 90-100 fps.

Thanks again deirn! i'll look around to finding a different mod for light source blocks as a work around for now or ill just keep it and just keep the Exotic Block blocks in vanilla chests outside of the network.

**Comment by deirn** [comment]
Spinnery uses a different item rendering method now, so that's probably the problem

**Comment by TheVillageid10t** [comment]
I reached out to Exotic Blocks' Discord and grondag took a look at it

Grondag 
I can fix this on my end but will be very inefficient.
Also, the problem is actually with the Spinnery library that SlotLink uses. @TheVillageIdiot
Well, Spinnery is not strictly incorrect - but it doesn't use the fabric API to allow for dynamic models so I'll have to do special handling to make it work.

They have it up on their own github as an issue as well. Havent tried their for now.

https://github.com/grondag/exotic-matter-2/issues/1

**Comment by grondag** [comment]
So I made a fix for this but now I can't reproduce the issue to confirm the fix.

**Comment by deirn** [comment]
still a problem on base spinnery, for now I'm using my own fork of v3.0.48 that still uses vanilla rendering

**Comment by grondag** [comment]
OK.  Exotic Matter build 343 on CF should fix it if you want to retest. Also on maven.  https://www.curseforge.com/minecraft/mc-mods/exotic-matter-library/files/3002624
